src/targets/targets.service.ts

```diff
@@ -31,12 +32,33 @@ export class TargetsService {
     longitude: number,
     userInfo: User,
     topicID: number,
-  ): Promise<Target> {
+  ): Promise<{ target: Target, matches: Target[] }> {
     const topic = await this.topicsRepository.findOne(topicID)
     const user = await this.usersRepository.findOne(userInfo)
 
-    const target = new Target(title, radius, latitude, longitude, user, topic)
-    return this.targetsRepository.save(target)
+    const sameTopicTargets = await this.targetsRepository.find({ topic })
+
+    const matches = sameTopicTargets.filter(target => {
+      const distance = getDistance(
+        { latitude: target.latitude, longitude: target.longitude },
+        { latitude, longitude }
+      )
+      return distance < target.radius + radius
+    })
+    
+    let target = new Target(title, radius, latitude, longitude, user, topic, matches)
+
+    for (const match of matches) {
```

Comment:
What are you trying to do here? There should be a higher-order function to do it anyway

Reply:
Regarding this, I'd rather have a `.forEach` if we are going to mutate since it transmits that intent. Performance shouldn't be the first thing to consider but how understandable the code is.